components/scream/src/share/util/scream_common_physics_functions_impl.hpp

```diff
@@ -208,6 +208,63 @@ void PhysicsFunctions<DeviceT>::calculate_z_int(const MemberType& team,
   column_ops::template column_scan<FromTop>(team,num_levs,dz,z_int,z_surf);
 }
 
+template<typename DeviceT>
+template<typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+ScalarT PhysicsFunctions<DeviceT>::calculate_vmr_from_mmr(const std::string& gas_name, const ScalarT& mmr)
+{
+  using C = scream::physics::Constants<Real>;
+  constexpr Real air_mol_weight   = C::MWdry;
+  constexpr auto& gas_mol_weights = C::gas_mol_weights;
+  EKAT_REQUIRE(gas_mol_weights.find(gas_name)!=gas_mol_weights.end());
+
+  return mmr / (1.0 - mmr) * air_mol_weight/gas_mol_weights.at(gas_name);
```

Comment:
This doesn't work on GPU, where you cannot use std::map. As I noted in another comment, the map will have to be replaced with something like a KOKKOS_FUNCTION, with some if-else inside that check the gas name (as a _C string_).